IPInfo.IO/ipinfo/trigger_fetch_ipinfo_database.py

```diff
@@ -0,0 +1,267 @@
+            )
+
+        # yield observables for IP segments
+        try:
+            result = [autonomous_system]
+
+            ip_start = ipaddress.ip_address(data["start_ip"])
+            ip_end = ipaddress.ip_address(data["end_ip"])
+
+            if ip_start.version != ip_end.version:
+                raise Exception("Ip start and end are not from the same version")
+            if ip_start.version not in (4, 6):
+                raise Exception("Only version 4 or 6 of IPs are supported")
+
+            observable_type = f"ipv{ip_start.version}-addr"
+            tags.append(
+                {
+                    "valid_from": tag_valid_from,
+                    "valid_until": tag_valid_until,
+                    "name": f"asn:{asn_number}",
+                }
+            )
+            for ip_range in ipaddress.summarize_address_range(ip_start, ip_end):
+                observable = self._create_observable(observable_type, ip_range, tags)
+                relationships = self._create_observable_relationship(
+                    observable, autonomous_system
+                )
+                result += [relationships, observable]
+
+            yield result
+        except Exception:
+            self.log(
+                message=f"Cannot parse provided ip addresses {row.decode()}",
+                level="error",
+            )
+
+    def _create_observable(
+        self,
+        observable_type: str,
+        ip_range: IPv4Network | IPv6Network,
+        tags: list,
+    ) -> dict:
+        return {
+            "type": observable_type,
+            "id": f"{observable_type}--{str(uuid.uuid4())}",
+            "value": str(ip_range),
+            "x_inthreat_tags": tags,
+            "x_inthreat_sources_refs": [self.identity["id"]],
+        }
+
+    def _create_observable_relationship(
+        self, observable: dict, autonomous_system: dict
+    ) -> dict:
+        return {
+            "id": f"observable-relationship--{uuid.uuid4()}",
+            "type": "observable-relationship",
+            "source_ref": observable["id"],
+            "target_ref": autonomous_system["id"],
+            "x_inthreat_sources_refs": [self.identity["id"]],
+            "relationship_type": "belongs-to",
+        }
+
+    def create_event_for_chunk(
+        self, location_chunk_info: tuple[list[dict], int]
+    ) -> None:
+        location_chunk = location_chunk_info[0]
+        offset = location_chunk_info[1]
+
+        # Add identity of the service to the bundle
+        location_chunk.append(self.identity)
+        work_dir = self._data_path.joinpath("ipinfo_chunks").joinpath(str(uuid.uuid4()))
+        chunk_path = work_dir.joinpath("observables.json")
+        work_dir.mkdir(parents=True, exist_ok=True)
+        with chunk_path.open("w") as fp:
+            fp.write(json.dumps(location_chunk))
+
+        directory = str(work_dir.relative_to(self._data_path))
+        file_path = str(chunk_path.relative_to(work_dir))
```

Comment:
The SDK now has helper to do this. it should be used.

See `write` method in `sekoia_automation.storage`

Reply:
Thank you for your review, weirdly the `write` from `sekoia_automation.storage` doesn't work, it keeps running until timeout. 
This is how I use it:
```python
    write(str(chunk_path), location_chunk)
```

Reply:
```suggestion
    file_path = write("observables.json", location_chunk)
    directory = file_path.parent.as_posix()
```

And in the `send_event`, replace `file_path=file_path` by `file_path="observables.json"`.